orc8r/cloud/go/services/directoryd/storage/storage_blobstore.go

```diff
@@ -121,6 +126,52 @@ func (d *directorydBlobstore) MapSessionIDsToIMSIs(networkID string, sessionIDTo
 	return store.Commit()
 }
 
+// MapHWIDToDirectoryRecordID stores mapping of {hwid -> directory record IDs}.
+func (d *directorydBlobstore) MapHWIDToDirectoryRecordIDs(networkID string, hwIDsToDirectoryRecordIDs map[string]*protos.DirectoryRecordIDs) error {
+	store, err := d.factory.StartTransaction(&storage.TxOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to start transaction")
+	}
+	defer store.Rollback()
+
+	blobs, err := convertDirectoryRecordIDsToBlobs(hwIDsToDirectoryRecordIDs)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal directory record IDs to blob values")
+	}
+	err = store.CreateOrUpdate(networkID, blobs)
+	if err != nil {
+		return errors.Wrap(err, "failed to create or update hwid to directory records mapping")
+	}
+	return store.Commit()
+}
+
+// GetDirectoryRecordIDsForHWID returns the directory record IDs mapped to
+// a hardwareID.
+func (d *directorydBlobstore) GetDirectoryRecordIDsForHWID(networkID string, hwid string) (*protos.DirectoryRecordIDs, error) {
+	store, err := d.factory.StartTransaction(&storage.TxOptions{ReadOnly: true})
```

Comment:
minor nit: Can you add some minor debug prints in case we want to debug any issue with greater verbosity?